NuRadioMC/utilities/medium_base.py

```diff
@@ -0,0 +1,336 @@
+    """
+    def __init__(self,z_airBoundary=0*units.meter,z_bottom=None):
+        self.z_airBoundary = z_airBoundary
+        self.z_bottom = z_bottom
+
+    def add_reflective_bottom(self,refl_z,refl_coef,refl_phase_shift):
+        """
+        function which adds a reflective bottom to your ice model
+        """
+        self.reflection = refl_z
+        self.reflection_coefficient = refl_coef
+        self.reflection_phase_shift = refl_phase_shift
+        
+        if not((self.z_bottom != None) and (self.z_bottom < self.reflection)):
+            self.z_bottom = self.reflection-1*units.m
+
+    def get_index_of_refraction(self,x):
+        """
+        returns the index of refraction at position x.
+        Reimplement everytime in the specific model
+
+        Parameters
+        ---------
+        x:  3dim np.array
+            point
+
+        Returns:
+        --------
+        n:  float
+            index of refraction
+        """
+        pass
+
+    def get_average_index_of_refraction(self,x1,x2):
+        """
+        returns the average index of refraction between two points
+        Reimplement everytime in the specific model
+
+        Parameters
+        ----------
+        x1: 3dim np.array
+            point
+        x2: 3dim np.array
+            point
+
+        Returns
+        -------
+        n_average:  float
+                    averaged index of refraction between the two points
+        """
+        pass
+
+    def get_gradient_of_index_of_refraction(self, x):
+        """
+        returns the gradient of index of refraction at position x
+        Reimplement everytime in the specific model
+
+        Parameters
+        ----------
+        x: 3dim np.array
+            point
+
+        Returns
+        -------
+        n_nabla:    (3,) np.array
+                    gradient of index of refraction at the point
+        """
+        pass
+
+    
+    def get_ice_model_radiopropa(self):
+        """
+        if radiopropa is installed this will return the ice model to insert in radiopropa
+
+        """
+        if radiopropa_is_imported:
+            # when implementing a new ice_model this part of the function should be ice model specific
+            pass
+        else:
+            logger.error('The radiopropa dependancy was not import and can therefore not be used. \nMore info on https://github.com/nu-radio/RadioPropa')
+            raise ImportError('RadioPropa could not be imported')
+
+
+class IceModel_Simple(IceModel):
+    """
+    predefined ice model (to inherit from) with exponential shape
+    """
+    def __init__(self,n_ice,delta_n,z_0,z_shift=0*units.meter,z_airBoundary=0*units.meter,z_bottom=None):
+        super().__init__(z_airBoundary,z_bottom)
+        self.n_ice = n_ice
+        self.delta_n = delta_n
+        self.z_0 = z_0
+        self.z_shift = z_shift
+
+    def get_index_of_refraction(self, x):
+        """
+        overwrite inherited function
+        """
```

Comment:
See comment on medium.py - in the current implementation, this would be the docstring the user sees for methods for any IceModel_Simple object.

Reply:
Ah yeah, I see you're point. I wanted to make obvious that a function is overwritten by this so the name and input and output should be the same, but maybe I can combine this with the docstring from the base_class